bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/model/commands/GetOperationmodeCommand.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands;
+
+import java.util.concurrent.Executor;
+
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaMessage;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaProperties.OperationMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+public class GetOperationmodeCommand extends BRC1HCommand {
+
+    private final Logger logger = LoggerFactory.getLogger(GetOperationmodeCommand.class);
+
+    private OperationMode operationMode;
+
+    @Override
+    public byte[] getRequest() {
+        return MadokaMessage.createRequest(this);
+    }
+
+    @Override
+    public boolean handleResponse(Executor executor, ResponseListener listener, MadokaMessage mm) {
+        try {
+            operationMode = OperationMode.valueOf(mm.getValues().get(0x20).getRawValue()[0]);
+
+            logger.debug("operationMode: {}", operationMode);
+
+            setState(State.SUCCEEDED);
+            executor.execute(() -> listener.receivedResponse(this));
+
+            return true;
+        } catch (Exception e) {
+            logger.debug("Error while parsing response", e);
+            setState(State.FAILED);
+        }
+        return false;
```

Comment:
It seems that your handler will log any failures when calling `handleResponse` already. So I think you would end up having duplicate logs in the event of exceptions in your commands.

Since the handler already handles any MadokaParsingException thrown, and since it appears that any exception here could be classified as a parsing exception, I suggest wrapping any exceptions that occur in a MadokaParsingException and passing the exception up the line.
Doing that would eliminate the double logging issue and would also eliminate the need to return a boolean since any failures would be thrown as an exception instead.
This should cleanup a bit of code in both the commands and the handler.
```suggestion
    public void handleResponse(Executor executor, ResponseListener listener, MadokaMessage mm) {
        try {
            operationMode = OperationMode.valueOf(mm.getValues().get(0x20).getRawValue()[0]);

            logger.debug("operationMode: {}", operationMode);

            setState(State.SUCCEEDED);
            executor.execute(() -> listener.receivedResponse(this));
        } catch (Exception e) {
            setState(State.FAILED);
            throw new MadokaParsingException(e);
        }
```